pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -199,14 +199,19 @@ public void preloadSegments(IndexLoadingConfig indexLoadingConfig) {
         return;
       }
       // From now on, the _isPreloading flag is true until the segments are preloaded.
+      long startTime = System.currentTimeMillis();
```

Comment:
We should use System.nanoTime to measure duration, since System.currentTimeMillis is not guaranteed to be monotonically increasing.

Reply:
I think doPreloadSegments method is coarse enough to offset the measurement error here. I'd use nanoTime when to measure very fine grained methods like those on the query execution path and those supposed to be finish in many ms or even under ms.

Reply:
It is not about precision rather correctness. https://pradeesh-kumar.medium.com/java-system-nanotime-vs-system-currenttimemillis-604f59555561 is a good read.